src/GraphControl/Control/Grapher.cpp

```diff
@@ -1091,6 +1099,10 @@ optional<vector<shared_ptr<Graphing::IEquation>>> Grapher::TryInitializeGraph(bo
         double xMin, xMax, yMin, yMax;
         m_graph->GetRenderer()->GetDisplayRanges(xMin, xMax, yMin, yMax);
         auto initResult = m_graph->TryInitialize(graphingExp);
+        if (IsKeepCurrentView)
+        {
+            m_resetUsingInitialDisplayRange = true;
+        }
         m_graph->GetRenderer()->SetDisplayRanges(xMin, xMax, yMin, yMax);
```

Comment:
Not this PR, but a crash will happen here if the above line `auto initResult = m_graph->TryInitialize(graphingExp);` fails to properly initialize the graph and returns nullopt to initResult.